provision/kubernetes/provisioner.go

```diff
@@ -746,6 +748,53 @@ func (p *kubernetesProvisioner) ListNodes(addressFilter []string) ([]provision.N
 	return nodes, nil
 }
 
+func (p *kubernetesProvisioner) InternalAddresses(ctx context.Context, a provision.App) ([]*provision.AppInternalAddress, error) {
+	client, err := clusterForPool(a.GetPool())
+	if err != nil {
+		return nil, err
+	}
+	ns, err := client.AppNamespace(a)
+	if err != nil {
+		return nil, err
+	}
+	addresses := []*provision.AppInternalAddress{}
+
+	tclient, err := TsuruClientForConfig(client.restConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	app, err := tclient.TsuruV1().Apps(client.Namespace()).Get(a.GetName(), metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	processes := []string{}
+	for process := range app.Spec.Services {
+		processes = append(processes, process)
+	}
+	sort.Strings(processes)
+
+	for _, process := range processes {
+		depName := deploymentNameForApp(a, process)
+		service, err := client.CoreV1().Services(ns).Get(depName, metav1.GetOptions{})
```

Comment:
Would you mind changing this to use the services informer?
It would look something like:
```
controller, err := getClusterController(p, client)
svcInformer, err := controller.getServiceInformer()
for ... {
	service, err := svcInformer.Lister().Services(ns).Get(depName)
```
We didn't update everywhere yet but for new code it's better to stick to informers since they won't cause another hit on the Kubernetes API and will instead read the service from a local cache that's being constantly updated in the background.

Reply:
Thanks so very much @cezarsa, I've applied all your suggestions. =)